PluginGenerator/JarManifestReader.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+ * SonarQube Roslyn SDK
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace SonarQube.Plugins
+{
+    /// <summary>
+    /// Reads a valid v1.0 Jar Manifest.
+    /// See http://docs.oracle.com/javase/6/docs/technotes/guides/jar/jar.html#JAR%20Manifest
+    /// </summary>
+    public class JarManifestReader
+    {
+        private readonly Dictionary<string, string> kvps;
+        private const string SEPARATOR = ": ";
+
+        public JarManifestReader(string manifestText)
+        {
+            if (manifestText == null)
+            {
+                throw new ArgumentNullException(nameof(manifestText));
+            }
+
+            kvps = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            // A manifest file line can have at most 72 characters. Long values are split across
+            // multiple lines, with the continuation line starting with a single space.
+            // The simplest way to rejoin the lines is just to replace all (EOL + space) with EOL
+            var joinedText = manifestText.Replace("\r\n ", string.Empty);
+            var lines = joinedText.Split(new string[] { "\r\n" }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Don't know why my previous comment was lost so I am redoing it:
This seems weird as you replace windows EOL by empty string and then on the replaced string you try to split on windows EOL.

Reply:
I deleted a line, which might by why your comment and my reply were lost.

See the comment in line 44: it's removing _EOL + space_  to re-join all of the continuation lines together. Any _EOL_ that are not followed by a space will still be there.

Reply:
OOOOOOOOH I didn't see the space after the `\r\n`... sorry for that.